src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -220,8 +222,92 @@ namespace
                     break;
                 }
             }
+
+            //TODO: update lights here
+        }
+        
+        void MaterialStructured(NodeKeyTab& nodes)
+        {
+            if (m_renderer == nullptr)
+                return;
+
+            for (int i = 0, e = nodes.Count(); i < e; ++i)
+            {
+                INode* node = NodeEventNamespace::GetNodeByKey(nodes[i]);
+                Mtl* mtl = node->GetMtl();
+                if (mtl == nullptr)
+                    continue;
+                m_renderer->assign_material(mtl, node);
+                m_renderer->get_render_session()->reininitialize_render();
```

Comment:
Is it OK to reinitialize the render at every iteration of the loop, or should it be done once outside the loop? Just asking.